### PR TITLE
Feature/infinite scroll

### LINKED ISSUE #26: [FE] Infinite Loading / Pagination Search Queries
~~This task requires discussion on how we want to handle changing the FE/BE json contract.~~
Update: We have decided that this will be handled purely in the front end due to keeping the rest of our features working. Specifically, sorting & filtering is currently being handled in the FE and will have issues if we only receive some data to sort at a time, as the order will be incorrect. This could be fixed by moving sorting and filtering to the backend, but has been put off to be a future task.

As a user, the results should come paginated, or infinitely loaded if given a high amount of results to reduce the amount of data being passed back and forward.

### PR DESCRIPTION
**Pull request by laurelbonal**
# Description

Added infinite scrolling to the search page and a "back to top" button

## Context

Used npm package to install infinite scrolling. Changes in SearchPage, CardDisplay, created a new component BackToTopButton, and cypress testing. 
please run ```npm install``` before reviewing code 

Fixes #26 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Added to cypress testing to ensure the back to top button works as expected

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by KojinKuro** [comment]
## Issue

`CardDisplay` infinite loading is not sorting correctly. If we search `a` we get the first 20 cards that is then sorted. After scrolling down, we get the first 40 card that is then sorted. This is not correct behavior. We should have all the cards sorted, then display the **first 20 of those cards sorted.** This is different than having the **first 20 cards which are then sorted**.

Expected Behavior: (sorting everything, then only display the first 20)
image

Current Behavior: (display the first 20, then sort)
image

## Other Notes
- Make sure you add some kind of Cypress test that checks to make sure infinite loading displays data correctly, matching the behavior described above.
- Should not need to pass the infinite loading props of `fetchMoreData` and `hasMore` to `CardDisplay`. These should be handled by CardDisplay internally as there is no reason why we are managing it in `SearchPage`. Any state related to infinite loading should be moved into `CardDisplay` for clarity instead of spreading it across 2 components making it hard to follow the flow of data.
- Move `BackTopButton` out of ui folder as all of those components are part of ShadCN

**Review by KojinKuro — CHANGES_REQUESTED** [review]
Update infinite loading to have correct sorting by sorting first, then performing infinite loading.

**Comment by laurelbonal** [comment]
> Update infinite loading to have correct sorting by sorting first, then performing infinite loading.

Updated infinite loading and also broke out some logic so that we have more reusability

**Comment by KojinKuro** [comment]
I tried looking up `a` in the search bar and it gave me all the cards at once 🤔 I looked through your code and couldn't find any infinite loading. Are you still planning on adding that?
